packages/grid/x-grid/src/DataGridPro.tsx

```diff
@@ -12,10 +12,10 @@ import {
   useGridApiRef,
```

Comment:
Can we also rename the folders? 

From `packages/grid/x-grid/src/XGrid.tsx` -> `packages/grid/data-grid-pro/src/DataGridPro.tsx`

Same for the `x-grid-data-generator` -> `data-grid-data-generator`.

Reply:
In #2313 there's a "Move the folders to be consistent with the main repository" item. I didn't move anything here because I thought it would be done in this task.

Reply:
That also works.